adal/src/main/java/com/microsoft/aad/adal/BasicWebViewClient.java

```diff
@@ -474,6 +476,40 @@ public void run() {
 
             openLinkInBrowser(parameters.get(INSTALL_URL_KEY));
             view.stopLoading();
+            return true;
+        } else if (url.toLowerCase(Locale.US).startsWith(PLAY_STORE_INSTALL_PREFIX)) {
```

Comment:
>else if [](start = 10, length = 7)

It looks like we have enough if-else cases here.
it might be a good idea to refactor this code to create separate UrlHandler instance based on the Url type with each handling the url in its own handleUrl method.

UrlType urlType = getUrlType(url);
switch (urlType)
{
case PLAYSTORE_INSTALL_URL:
return PlayStoreUrlHandler.HandleUrl(url);
case AUTHENTICATOR_MFA_LINKING:
return AuthenticatorMFALinkingUrlHandler.HandlerUrl(url);
...
}

This way the code can be shared between MSAL and ADAL, rather than duplicating, right?

Reply:
The reason why this extensive refactoring isnt done is because, unlike MSAL, ADAL does not depend on common module when it comes to handling Overriden Url loading, also ADAL is on the path to be deprecated, and would be having far less releases, so at this point it wouldnt be worth refactoring this entire logic for ADAL, and there might be few usecases involved here which would not have been exactly mirrored as MSAL. Hence for quick resolution, we have added the capability of supporting 1st party brokers in the existing code of ADAL.